proposals/0000-existentials.rst

```diff
@@ -221,40 +243,54 @@ Proposed Change Specification
 
    1. When checking an expression ``e`` against a type ``exists (a :: k). ty``, we create a fresh unification
       variable ``α :: k`` and check ``e`` against ``ty[α/a]``. (This is reflected in rule ``Gen`` at the top
-      of Fig. 4 of the paper_.)
+      of Fig. 4 of the paper_.) Here is the rule, with desugaring into Core (referencing new Core constructs
+      introduced below)::
+
+        (α :: k) fresh
+        Γ ⊢ e <= ty[α/a] ~> e'
+        -----------------------------------------------
+        Γ ⊢ e <= exists (a :: k). ty ~> Pack α e' a ty
 
    #. When checking an expression ``e`` against a type ``C /\ ty``, continue by checking ``e`` against ``ty``,
       emitting ``C`` as a wanted. That is, we must be able to satisfy the constraint ``C`` in order to accept
-      ``e`` as type ``C /\ ty``. (This is reflected in rule ``GenQualified`` in Fig. 11 of the paper_.)
+      ``e`` as type ``C /\ ty``. (This is reflected in rule ``GenQualified`` in Fig. 11 of the paper_.) ::
+
+        Γ ⊩ C ~> dict
+        Γ ⊢ e <= ty ~> e'
+        ----------------------------------------
+        Γ ⊢ e <= C /\ ty ~> Mk/\ @C @ty dict e'
 
    #. When inferring the type of an expression ``e``, if that type is ``exists (a :: k). ty``, then rewrite
       that type to become ``ty[Witness @k (exists (a :: k). ty) e/a]``. (This is rule ``IExist`` in Fig. 5
-      of the paper_.)
+      of the paper_.) ::
+
+        Γ ⊢ e => exists (a :: k). ty ~> e'
+        -------------------------------------------
+        Γ ⊢ e => ty[Witness @k ty e / a] ~> Open e'
 
    #. When inferring the type of an expression ``e``, if that type is ``C /\ ty``, then rewrite that
       type to become ``ty`` and assume ``C`` as a given, available for use anywhere in the innermost
       case-match or lambda, including invisible lambdas as introduced in the argument to a function
       with a higher-rank type. (This is rule ``IGiven`` in Fig. 11 of the paper_.) Use of such a given
       may have a surprising influence on runtime behavior, see Ambiguity_, below.
 
-   #. When inferring the type of a lambda-expression ``\ pats -> e``, we must ensure that no variable bound
-      in ``pats`` leaks into the type of the expression; this is like a skolem-escape.
+        Γ ⊢ e => C /\ ty ~> e'   -- adding C to the environment not modeled here
+        ----------------------
+        Γ ⊢ e => ty ~> sndC e'
```

Comment:
I believe you want to add a double colon to the end of the top line, like so:

```rst
may have a surprising influence on runtime behavior, see Ambiguity_, below. ::

   Γ ⊢ e => C /\ ty ~> e'   -- adding C to the environment not modeled here
        ----------------------
   Γ ⊢ e => ty ~> sndC e'
```

Disclaimer: I don't know much about `rst`, but the latest commit wasn't rendering on GitHub for me, and making the above change fixed it.

Reply:
Yes -- thank you!

Reply:
You're welcome; thanks for all your work!